c++/src/kj/common.h

```diff
@@ -373,6 +373,23 @@ struct DisallowConstCopy {
   DisallowConstCopy& operator=(DisallowConstCopy&&) = default;
 };
 
+#if _MSC_VER
+
+#define MSVC_CAPTURE(obj) obj=::kj::cp(obj)
```

Comment:
It's nice to hear that someone understands and likes my unconventional style choices. :)

However, we've had many other problems with the non-const copy approach. The Clang/GCC ABI incompatibility is pretty severe. So it should probably go. C++ just isn't going to be Rust any time soon, I guess.